### ISSUE TITLE
Integration test failing with version 3.0.0 and Postgres > 9.6

### ISSUE THREAD

**Issue by marcschumacher** [zalando-stups/java-sproc-wrapper#72]
I am running the latest version of Postgres from Docker with `docker run -d --name sprocwrapper-postgres13 -e POSTGRES_USER=postgres -e POSTGRES_PASSWORD=postgres -p 5432:5432 postgres:13`.

Now starting the integration tests with `mvn clean verify -Pintegration-test` cause the test `SimpleIT.testReturnDomainObjectWithNullInnerObject:848` to fail:

```
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.014 s - in org.zalando.sprocwrapper.PartialDatabaseTypeIT
[INFO] Running org.zalando.sprocwrapper.ShardingIT
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.624 s - in org.zalando.sprocwrapper.ShardingIT
[INFO] Running org.zalando.sprocwrapper.ParallelExecutionIT
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.897 s - in org.zalando.sprocwrapper.ParallelExecutionIT
[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   SimpleIT.testReturnDomainObjectWithNullInnerObject:848
[INFO]
[ERROR] Tests run: 206, Failures: 1, Errors: 0, Skipped: 3
```

**Comment by marcschumacher** [comment]
@vadeg Can you help with that?

**Comment by vadeg** [comment]
@marcschumacher this happens because Postgres have changed the behaviour how to return empty composite types. The test executes the following query
```
select * from ztest_schema1.get_entity_with_null_inner_object();
```
but the returned results are different for different Postgres instances

### Postgres 9

```
local_test_db1=# select * from ztest_schema1.get_entity_with_null_inner_object();
 a |  b  | c
---+-----+---
 a | (,) |
(1 row)
```

The result contains an object in the field `b` but with all empty values inside.

### Postgres 13

```
local_test_db1=# select * from ztest_schema1.get_entity_with_null_inner_object();
 a | b | c
---+---+---
 a |   |
(1 row)
```

The same query returns `null` for  `b` instead of empty object as before.

The test returns the following object instance
```
public class ExampleDomainObjectWithInnerObject {
    @DatabaseField(name = "a")
    public String a;

    @DatabaseField(name = "b")
    public ExampleDomainObject b;

    @DatabaseField(name = "c")
    public List<ExampleDomainObject> c;
```

For Postgres 9 the field `b` will contain an object with all fields initialized as `null` or as default values in case of primitives.
For Postgres 13 the field `b` will contain `null`, because the response from Postgres returns `null`.

I think that behaviour with Postgres 13 is a more safe behaviour which prevents creating graphs of objects with empty / `null` values, while the behaviour with Postgres 9 looks more like a bug, which can lead to a bigger (slightly) memory footprint usage. 

@Jan-M maybe you have any hints why it was implemented like that?

**Comment by Jan-M** [comment]
@CyberDem0n FYI if that is true that may need some care when upgrading pg9.6 to pg13 where sprocs are used.

**Comment by Jan-M** [comment]
I think the test was implemented against 9.6 result to verify it works and stays as expected.

Now with the new jdbc driver and pg13 and behavior that arguably looks correct we just need to adapt the integration test to this.

**Comment by vadeg** [comment]
Fixed in #85
